File: April_13_LoadingAnimation/April_13_LoadingAnimation/ViewController.m
//
//  ViewController.m
//  April_13_LoadingAnimation
//
//  Created by HuangGuoQing on 16/4/13.
//  Copyright © 2016年 HJL. All rights reserved.
//

#define SCREENWIDTH [UIScreen mainScreen].bounds.size.width

#import "ViewController.h"

@interface ViewController () <UITableViewDataSource, UITableViewDelegate> {
    NSInteger n;
}

@property (nonatomic)UITableView *tableView;

@end

@implementation ViewController

- (UITableView *)tableView {
    if (!_tableView) {
        _tableView = [[UITableView alloc] initWithFrame:CGRectMake(0, 100, SCREENWIDTH, 400)];
        _tableView.delegate = self;
        _tableView.dataSource = self;
        n = 5;
    }
    return _tableView;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.view addSubview:self.tableView];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return n;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:nil];
    cell.textLabel.text = [NSString stringWithFormat:@"第%ld个cell", indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView beginUpdates];
    [tableView insertRowsAtIndexPaths:@[[NSIndexPath indexPathForRow:1 inSection:0]] withRowAnimation:UITableViewRowAnimationTop];
    n++;
    [tableView endUpdates];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: April_12/April_12/ViewController.swift
//
//  ViewController.swift
//  April_12
//
//  Created by HuangGuoQing on 16/4/12.
//  Copyright © 2016年 HJL. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var myLabel:UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = UIColor .cyanColor();
        
        myLabel = UILabel();
        
        myLabel.text = "这是一个label";
        myLabel.frame = CGRectMake(30, 100, 200, 30);
        self.view .addSubview(myLabel);
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

